### PR TITLE
[Feature] Guardian/POA Form Validation Refactoring

### PR DESCRIPTION
**Pull request by bonnie-chin**
## Notion ticket link

[Guardian/POA Form Validation](https://www.notion.so/uwblueprintexecs/Guardian-POA-Form-Validation-3de7c3603c614f87af18bb818fcc7f08)

## Implementation description
* Modelled after payment information form validation for conditional form validation
* Created Yup validation schema
* Refactored form fields with Formik wrapper and input fields
* Added validation schema and initial values to Formik wrappers in forms

image

image

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

File: lib/guardian/validation.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+export const guardianInformationSchema = object({
+  omitGuardianPoa: bool().default(false),
+  firstName: string()
+    .nullable()
+    .default(null)
+    .when('omitGuardianPoa', {
+      is: false,
+      then: string()
+        .typeError('Please enter a first name')
+        .matches(/^[a-zA-Z]*$/, 'Must only contain letters')
```
Comment (angeladietz) [inline lib/guardian/validation.ts:15]: I think we can remove the matches line. For instance, this won't allow names with hyphens. Even though other non-alpha characters are pretty unlikely to be in names I think it's okay to allow it to be any string which is how we're handling this in other forms. Same goes for middle name, last name, and city.

**Review by angeladietz — CHANGES_REQUESTED** [review]
Looks awesome! Just left a couple comments

File: lib/guardian/validation.ts (hunk 1 above)
Comment (bonnie-chin) [reply lib/guardian/validation.ts:15]: Good point, I didn't think about the hyphens :)

File: components/admin/requests/guardian-information/Form.tsx [hunk 2]
```diff
@@ -77,21 +62,23 @@ export default function GuardianInformationForm({
...
-            <TextField name="addressLine1" label="Address Line 1" required={true}>
+            <TextField name="guardianInformation.addressLine1" label="Address line 1" required>
               <FormHelperText color="text.secondary">
                 {'Street Address, P.O. Box, Company Name, c/o'}
               </FormHelperText>
             </TextField>
-            <TextField name="addressLine2" label="Address Line 2">
-              {/* TODO: add optional prop for (optional) in header */}
+            <TextField name="guardianInformation.addressLine2" label="Address Line 2">
+              <Box as="span" textStyle="caption" fontSize="sm">
```
[the remark this answers is not in the document]
Comment (bonnie-chin) [reply components/admin/requests/guardian-information/Form.tsx:71]: whoaaa didn't know you could pass jsx elements like that damn 👀

**Review by angeladietz — APPROVED** [review]
🚀

**Pull request merged**